From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 3 Aug 2010 13:51:43 -0400
Subject: [net] netxen: fix deadlock in aer
Message-id: <20100803134859.2239.18467.sendpatchset@localhost.localdomain>
Patchwork-id: 27337
O-Subject: [RHEL 5.6 PATCH 35/44] netxen: fix deadlock in aer
Bugzilla: 562937
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Bugzilla
========

562937

Upstream Status
===============

Not upstream yet.

Description
===========

>From 43cbeb05f6d3d9066b3c56691b3c286b08e72709 Mon Sep 17 00:00:00 2001
From: Amit Kumar Salecha <amit.salecha@qlogic.com>
Date: Thu, 8 Apr 2010 03:53:27 -0700
Subject: [PATCH] netxen: fix deadlock in aer

Aer uses kernel workqueue and we are calling flush_scheduled_work
in detach_func during aer recovery.

Signed-off-by: Amit Kumar Salecha <amit.salecha@qlogic.com>

diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index 888454e..046054c 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -1480,8 +1480,6 @@ static void netxen_nic_detach_func(struct netxen_adapter *adapter)
 	if (netif_running(netdev))
 		netxen_nic_down(adapter, netdev);
 
-	flush_scheduled_work();
-
 	netxen_nic_detach(adapter);
 
 	if (adapter->portnum == 0)
